synapse/config/registration.py

```diff
@@ -33,6 +33,9 @@ def read_config(self, config):
 
         self.registrations_require_3pid = config.get("registrations_require_3pid", [])
         self.allowed_local_3pids = config.get("allowed_local_3pids", [])
+        self.check_is_for_allowed_local_3pids = config.get(
+            "check_is_for_allowed_local_3pids", False
```

Comment:
Given this is a string rather than boolean, I'd just have it default to None here. Don't think you need to change any other code.